### DISCUSSION TITLE
integrating webpack resolve.modules with pnp

### DISCUSSION THREAD

**Discussion by qhenkart** [yarnpkg/berry#4159] [Q&A] [answered]
I use a similar pattern from the webpack documentation so that I can traverse my components like modules. Instead of using ../../../path/to/component. I just do path/to/component.

I am using yarn@latest and hoping to migrate to Plug n Play (pnp). I am wondering if there is a way to migrate this configuration so I can still take advantage of it. Possibly a way to require.resolve all modules?

```
const path = require('path');

module.exports = {
  //...
  resolve: {
    modules: [path.resolve(__dirname, 'src'), 'node_modules'],
  },
};
```

**Comment by arcanis** [comment, marked as answer]
Cf https://yarnpkg.com/features/protocols#why-is-the-link-protocol-recommended-over-aliases-for-path-mapping

**Reply by qhenkart** [reply]
as the documentation says. Turns out Yarn supports this automatically without further configuration.

**Comment by tire9876** [comment]
you can integrate from [tire shop](https://atireshop.com/) website.

**Comment by chairshop** [comment]
the webpack is available at [chair shop](https://bestchairshop.com/) website
